ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -777,13 +777,14 @@ public ValidTxnList getValidTxns(List<TxnType> excludeTxnTypes) throws LockExcep
   public ValidTxnWriteIdList getValidWriteIds(List<String> tableList,
                                               String validTxnList) throws LockException {
     assert isTxnOpen();
-    assert validTxnList != null && !validTxnList.isEmpty();
-    try {
-      return TxnCommonUtils.createValidTxnWriteIdList(
-          txnId, getMS().getValidWriteIds(tableList, validTxnList));
-    } catch (TException e) {
-      throw new LockException(ErrorMsg.METASTORE_COMMUNICATION_FAILED.getMsg(), e);
+    if (!StringUtils.isEmpty(validTxnList)) {
```

Comment:
Why do we need this change ? Please don't mix new code/feature with refactoring, as it makes it harder to review.

Reply:
When ever validTxnList is null Assert will return false and break the entire execution loop. Due to which 40 odd UTs are failing. Hence I have to replace the assert check to simple null check to get the green build.
for list of test failing please check - http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-2211/7/tests